gluoncv/model_zoo/icnet.py

```diff
@@ -0,0 +1,381 @@
+from __future__ import division
+from mxnet.gluon import nn
+from mxnet.context import cpu
+from mxnet.gluon.nn import HybridBlock
+from gluoncv.model_zoo.segbase import SegBaseModel
+from gluoncv.model_zoo.pspnet import _PSPHead
+from gluoncv.loss import SoftmaxCrossEntropyLoss
+
+
+__all__ = ['ICNet', 'get_icnet', 'get_icnet_resnet50_citys', 'ICNetLoss']
+
+
+class ICNetLoss(SoftmaxCrossEntropyLoss):
+    """Weighted SoftmaxCrossEntropyLoss2D
+
+    Parameters
+    ----------
+    weights : tuple, default (0.4, 0.4, 0.4)
+        The weight for cascade label guidance.
+    """
+    def __init__(self, weights=(0.4, 0.4, 0.4), **kwargs):
+        super(ICNetLoss, self).__init__(**kwargs)
+        self.weights = weights
+
+    def _weighted_forwarad(self, F, scale_pred1, scale_pred2,
+                           scale_pred3, scale_pred4, label, **kwargs):
+        h, w = label.shape[1], label.shape[2]
+
+        scale_pred = F.contrib.BilinearResize2D(scale_pred1, height=h, width=w)
+        loss1 = super(ICNetLoss, self).hybrid_forward(F, scale_pred, label)
+
+        scale_pred = F.contrib.BilinearResize2D(scale_pred2, height=h, width=w)
+        loss2 = super(ICNetLoss, self).hybrid_forward(F, scale_pred, label)
+
+        scale_pred = F.contrib.BilinearResize2D(scale_pred3, height=h, width=w)
+        loss3 = super(ICNetLoss, self).hybrid_forward(F, scale_pred, label)
+
+        scale_pred = F.contrib.BilinearResize2D(scale_pred4, height=h, width=w)
+        loss4 = super(ICNetLoss, self).hybrid_forward(F, scale_pred, label)
+
+        return loss1 + self.weights[0] * loss2 + \
+               self.weights[1] * loss3 + self.weights[2] * loss4
+
+    def hybrid_forward(self, F, *inputs, **kwargs):
+        """Compute loss"""
+        return self._weighted_forwarad(F, *inputs, **kwargs)
+
+
+class ICNet(SegBaseModel):
+    r"""Image Cascade Network (ICNet)
+
+    Parameters
+    ----------
+    nclass : int
+        Number of categories for the training dataset.
+    backbone : string
+        Pre-trained dilated backbone network type (default:'resnet50'; 'resnet50',
+        'resnet101' or 'resnet152').
+    norm_layer : object
+        Normalization layer used in backbone network (default: :class:`mxnet.gluon.nn.BatchNorm`;
+    norm_kwargs : dict
+        Additional `norm_layer` arguments, for example `num_devices=4`
+        for :class:`mxnet.gluon.contrib.nn.SyncBatchNorm`.
+    pretrained_base : bool or str
+        Refers to if the FCN backbone or the encoder is pretrained or not. If `True`,
+        model weights of a model that was trained on ImageNet is loaded.
+
+
+    Reference:
+
+        Hengshuang Zhao, Xiaojuan Qi, Xiaoyong Shen, Jianping Shi and Jiaya Jia.
+        "ICNet for Real-Time Semantic Segmentation on High-Resolution Images." *ECCV*, 2018
+
+    Examples
+    --------
+    >>> model = ICNet(nclass=19, backbone='resnet50')
+    >>> print(model)
+    """
+    # pylint: disable=arguments-differ
```

Comment:
I think you already disable arguments-differ in the beginning of the file, there is no need to disable it again. Please remove this line.